api/C/include/csi_ll.h

```diff
@@ -36,5 +38,37 @@ typedef enum {
  */
 unsigned csi_ll_set_interrupt_enables(unsigned int_enables);
 
+/*
+ * Set the frequency of the system timer.  Note that there is typically a single
+ * timer for all harts in the system, so this function can affect the operation of
+ * other harts.  When using the RVM-CSI high-level interrupt and timer support
+ * module, the timer will be compared against a compare register for each hart in
+ * order to produce a regular timer interrupt at a tick rate configured using
+ * csi_set_timer_tick, which is used for timing purposes.  This function must run
+ * in machine mode.
+ *
+ * @param timer_freq_mhz: System timer frequency in MHz
+ * @return : Status of operation.  CSI_ERROR will be returned if the request is
+ * invalid.
+ */
+csi_status_t csi_timer_config(unsigned timer_freq_mhz);
```

Comment:
As above

Reply:
I don't think the timer_freq_mhz argument has to be unsigned long, as 32-bits is more than enough to express a timer frequency in MHz.  This value does not have to match a register width, as in general it will not be directly programmed into a register.